src/NHibernate/Async/Action/BulkOperationCleanupAction.cs

```diff
@@ -57,37 +59,46 @@ public Task ExecuteAsync(CancellationToken cancellationToken)
 			}
 		}
 
-		private async Task EvictCollectionRegionsAsync(CancellationToken cancellationToken)
+		private Task EvictCollectionRegionsAsync(CancellationToken cancellationToken)
 		{
-			cancellationToken.ThrowIfCancellationRequested();
-			if (affectedCollectionRoles != null)
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<object>(cancellationToken);
+			}
+			try
 			{
-				foreach (string roleName in affectedCollectionRoles)
+				if (affectedCollectionRoles != null && affectedCollectionRoles.Any())
 				{
-					await (session.Factory.EvictCollectionAsync(roleName, cancellationToken)).ConfigureAwait(false);
+					return session.Factory.EvictCollectionAsync(affectedCollectionRoles, cancellationToken);
 				}
+				return Task.CompletedTask;
+			}
+			catch (Exception ex)
+			{
+				return Task.FromException<object>(ex);
 			}
 		}
 
-		private async Task EvictEntityRegionsAsync(CancellationToken cancellationToken)
+		private Task EvictEntityRegionsAsync(CancellationToken cancellationToken)
 		{
-			cancellationToken.ThrowIfCancellationRequested();
-			if (affectedEntityNames != null)
+			if (cancellationToken.IsCancellationRequested)
 			{
-				foreach (string entityName in affectedEntityNames)
+				return Task.FromCanceled<object>(cancellationToken);
+			}
+			try
+			{
+				if (affectedEntityNames != null && affectedEntityNames.Any())
 				{
-					await (session.Factory.EvictEntityAsync(entityName, cancellationToken)).ConfigureAwait(false);
+					return session.Factory.EvictEntityAsync(affectedEntityNames, cancellationToken);
 				}
+				return Task.CompletedTask;
+			}
+			catch (Exception ex)
+			{
+				return Task.FromException<object>(ex);
 			}
 		}
 
 		#endregion
-
-		public virtual async Task InitAsync(CancellationToken cancellationToken)
```

Comment:
"Since the class is not internal, this is a breaking change. It needs to be added back, obsoleted, in the non-generated partial."